components/feature/prompts/src/main/java/mozilla/components/feature/prompts/PromptFeature.kt

```diff
@@ -816,6 +816,13 @@ class PromptFeature private constructor(
         dialog.feature = this
 
         if (canShowThisPrompt(promptRequest)) {
+            // If the active prompt dialog has been updated, then dismiss it
+            activePrompt?.get()?.let { promptDialog ->
+                if (promptDialog.promptRequestUID !in session.content.promptRequests.map { prompt -> prompt.uid }) {
+                    promptDialog.dismiss()
```

Comment:
As we are reusing the geckoView callbacks in the updated prompt, we have to be careful of not dismissing them, I they can be used only once , when we when we dismiss the dialog fragment are we also this dismissing the GeckoView callbacks?

https://github.com/mozilla-mobile/android-components/blob/main/components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/ChoiceDialogFragment.kt#L106-L130

Reply:
What do you think about this approach? :)  
On `PromptFeature.kt`

```kotlin 
    internal fun dismissPreviousDialogIfNeeded(promptRequest: PromptRequest) {
        activePrompt?.get()?.let { promptDialog ->
            // ChoiceDialogFragment could update their choices data, 
            // and we need to dismiss the previous UI dialog,
            // without consuming the engine callbacks, allow to create a new dialog with the
            // update data.
            if (promptDialog.promptRequestUID == promptRequest.uid && promptDialog is ChoiceDialogFragment) {
                promptDialog.shouldCancelFeatureOnDismiss = false
                promptDialog.dismiss()
            }
        }
    }
```

On `ChoiceDialogFragment.kt`

```kotlin
internal class ChoiceDialogFragment : PromptDialogFragment() {

    internal var shouldCancelFeatureOnDismiss = true

...
...
...
}
```

Reply:
Thank you for your suggestion! I added it to the PR.